addons/mail/static/src/web/chat_window/chat_window_service.js

```diff
@@ -152,7 +152,7 @@ export class ChatWindowService {
             // ensure messaging menu is opened before chat window is closed
             await Promise.resolve();
         }
-        if (this.maxVisible < this.store.chatWindows.length) {
+        if (this.maxVisible < this.store.chatWindows.length && !chatWindow.hidden) {
```

Comment:
I would put condition at the front, so it's more apparent that this branching is to handle swapping on visible chat windows